dsymbol/src/dsymbol/ufcs.d

```diff
@@ -21,14 +21,20 @@ enum CompletionContext
     ParenCompletion,
 }
 
+
 struct TokenCursorResult
 {
     CompletionContext completionContext;
     istring functionName;
-    istring symbolIdentifierName;
+    Token significantToken;
     string partialIdentifier;
 }
 
+struct DeducedSymbolTypeResult{
+    const(DSymbol)* deducedSymbolType;
+    bool success = false;
```

Comment:
can success ever be true when the pointer is null or ever be false when the pointer is not null?

It'd probably make sense to just return a DSymbol and check if it's null or not to see if it's succeeded.